			   READLINE PATCH REPORT
			   =====================

Readline-Release: 8.2
Patch-ID: readline82-007

Bug-Reported-by:	Kevin Pulo <kev@pulo.com.au>
Bug-Reference-ID:
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-readline/2022-11/msg00002.html

Bug-Description:

If readline is called with no prompt, it should display a newline if return
is typed on an empty line. It should still suppress the final newline if
return is typed on the last (empty) line of a multi-line command.

Patch (apply with `patch -p0'):

*** ../readline-8.2-patched/display.c	2022-04-05 10:47:31.000000000 -0400
--- display.c	2022-12-13 13:11:22.000000000 -0500
***************
*** 3342,3348 ****
  		 &last_face[_rl_screenwidth - 1 + woff], 1);
      }
!   _rl_vis_botlin = 0;
!   if (botline_length > 0 || _rl_last_c_pos > 0)
      rl_crlf ();
    fflush (rl_outstream);
    rl_display_fixed++;
--- 3394,3400 ----
  		 &last_face[_rl_screenwidth - 1 + woff], 1);
      }
!   if ((_rl_vis_botlin == 0 && botline_length == 0) || botline_length > 0 || _rl_last_c_pos > 0)
      rl_crlf ();
+   _rl_vis_botlin = 0;
    fflush (rl_outstream);
    rl_display_fixed++;
*** ../readline-8.2/patchlevel	2013-11-15 08:11:11.000000000 -0500
--- patchlevel	2014-03-21 08:28:40.000000000 -0400
***************
*** 1,3 ****
  # Do not edit -- exists only for use by patch
  
! 6
--- 1,3 ----
  # Do not edit -- exists only for use by patch
  
! 7
